src/test/kotlin/org/malachite/estella/people/JobSeekerServiceTest.kt

```diff
@@ -0,0 +1,134 @@
+package org.malachite.estella.people
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.malachite.estella.commons.models.people.JobSeeker
+import org.malachite.estella.services.JobSeekerService
+import org.malachite.estella.services.MailService
+import org.malachite.estella.services.UserService
+import org.malachite.estella.util.jobSeekers
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.http.client.reactive.ClientHttpRequest
+import org.springframework.util.MultiValueMap
+import org.springframework.web.reactive.function.client.ClientResponse
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.util.context.Context
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+import java.nio.charset.Charset
+import java.time.ZonedDateTime
+import java.util.function.Consumer
+import java.util.function.Function
+
+class JobSeekerServiceTest {
+
+    private val jobSeekerRepository = DummyJobSeekerRepository()
+    private val userRepository = DummyUserRepository()
+    private val mailServiceMock = mockk<MailService>()
+    private val userService = UserService(userRepository, mailServiceMock)
+    private val service = JobSeekerService(jobSeekerRepository, userService)
+
+    @BeforeEach
+    fun setup() {
+        every { mailServiceMock.sendRegisterMail(any()) } returns mailServiceResponse
+    }
+
+    @AfterEach
+    fun cleanup() {
+        jobSeekerRepository.clear()
+        userRepository.clear()
+    }
+
+    @Test
+    fun `should be able to add job seekers`() {
+        testUsers.forEach {
+            service.addJobSeeker(it)
+        }
+        expectThat(service.getJobSeekers()).isEqualTo(testUsers as MutableIterable<JobSeeker>)
+    }
+
+    @Test
+    fun `should be able to register jobSeekers and send mails to them`() {
+        testUsers.forEach {
+            service.registerJobSeeker(it)
+        }
+        expectThat(service.getJobSeekers()).isEqualTo(testUsers as MutableIterable<JobSeeker>)
+        verify(exactly = testUsers.size) { mailServiceMock.sendRegisterMail(any()) }
+    }
+
+    private val testUsers = jobSeekers
+
+}
+
+private val mailServiceResponse = object: WebClient.RequestHeadersSpec<Nothing> {
```

Comment:
Yyyy to przed zamknięciem PR'a będzie zaimplementowane, czy też osobny task na to? I czy nie dałoby się tego wyrzucić do osobnego pliku bo strasznie zmniejsza czytelność wg mnie

Reply:
ogólnie chciałem to jakoś ładnie zrobić, ale jak mockujesz, to w tej bibliotece każde wywołanie metody na mocku musi być napisane, co zwraca. Jako że jest to jakieś gówno z WebClienta, to nie mogę dać ani Any, ani Unit, ani nic normalnego. To jest po prostu taka mockowa, nie potrzebna nikomu odpowiedź, tyle tylko, że jest zwracana przez MailService. Ogólnie mogę się tego pozbyć, ale wtedy też pozbywam się weryfikacji, czy metoda została wywołana ileś razy.

Reply:
A nie da się chociaż przenieść tego do innego pliku ?

Reply:
jakby się to dało gdzieś wynieść na /dev/null i tam więcej nie patrzeć to by było super xD

Reply:
zgaduję, że nazwa `dev/shits/dontLookHere` nie przejdzie? XD